lib/internal/errors.js

```diff
@@ -954,6 +953,20 @@ E('ERR_TRANSFORM_ALREADY_TRANSFORMING',
 E('ERR_TRANSFORM_WITH_LENGTH_0',
   'Calling transform done when writableState.length != 0', Error);
 E('ERR_TTY_INIT_FAILED', 'TTY initialization failed', SystemError);
+E('ERR_TYPE_MISMATCH', (filename, ext, typeFlag, conflict) => {
```

Comment:
If this is only intended to be used for ES modules, then I'd prefer that we add `_ESM_` to the error code.

Reply:
It also applies to CommonJS, e.g. `node --entry-type=module file.cjs` or `node --entry-type=commonjs file.mjs`.

Reply:
@TimothyGu is is ok to drop this request based on response?